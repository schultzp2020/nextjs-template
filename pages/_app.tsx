import 'tailwindcss/tailwind.css';
import { Layout } from '@src/components';
import type { AppProps } from 'next/app';

const App: React.FC<AppProps> = ({ Component, pageProps }: AppProps) => (
  <Layout>
    <Component {...pageProps} />
  </Layout>
);

export default App;
