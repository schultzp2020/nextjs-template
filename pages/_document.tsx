import { Html, Head, Main, NextScript } from 'next/document';

function MyDocument(): JSX.Element {
  return (
    <Html lang="en-US">
      <Head />
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}

export default MyDocument;
