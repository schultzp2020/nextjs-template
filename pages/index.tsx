import type { NextPage } from 'next';
import { JsonSchemaViewer } from '@stoplight/json-schema-viewer';
import { schema } from '../devfileJsonSchema';

export const Home: NextPage = () => (
  <div>
    <JsonSchemaViewer schema={schema} emptyText="No schema defined" defaultExpandedDepth={0} />
  </div>
);

export default Home;
