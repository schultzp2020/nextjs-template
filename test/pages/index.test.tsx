import React from 'react';
import { render } from '../testUtils';
import { Home } from '../../pages/index';
import { describe, it, expect } from '@jest/globals';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
