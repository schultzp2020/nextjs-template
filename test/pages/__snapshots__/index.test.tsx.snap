// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <p>
      Hello World!
    </p>
  </div>
</DocumentFragment>
`;
